### PR TITLE
feat(scan): write-in adjudication

### PR DESCRIPTION
**Pull request by eventualbuddha**
Still needs a lot of work:
- [x] well-factored code (deferred to #850)
- [x] CSS/UI fixes? (sorry @beausmith) (deferred to #849)
- [x] tests
- [x] cleanup any partially-implemented bits & leftovers from experiments
- [x] FEAT: autocomplete for candidate names (deferred to #846)
- [x] FEAT: ability to "cancel" a write-in (deferred to #847)
- [x] BUG: briefly shows the adjudication screen after saving and showing the main dashboard (deferred to #848)

Try it out locally by adding `WriteIn` as an enabled adjudication reason:

```sh
❯ sqlite3 -interactive dev-workspace/ballots.db
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
sqlite> update configs set value = (select json_set(value, '$.election.precinctScanAdjudicationReasons', json_array('UninterpretableBallot','Overvote','Undervote','BlankBallot','WriteIn')) from configs where key = 'election' limit 1) where key = 'election';sqlite> update configs set value = (select json_set(value, '$.election.centralScanAdjudicationReasons', json_array('UninterpretableBallot','Overvote','Undervote','BlankBallot','WriteIn')) from configs where key = 'election' limit 1) where key = 'election';
```

Refs #802

### CODE REVIEW

File: apps/bsd/src/screens/WriteInAdjudicationScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+const ContestAdjudication = ({
+  imageURL,
+  contest,
+  layout,
+  writeInsForContest,
+  onInput,
+  writeInValues,
+}: ContestAdjudicationProps): JSX.Element => {
+  const onInputInternal: React.FormEventHandler = useCallback(
+    (event) => {
```
Comment (carolinemodic) [inline apps/bsd/src/screens/WriteInAdjudicationScreen.tsx:188]: nit: I think we can type event here which can probably let you avoid the as HTMLInputElement below but I'm not sure what the right type is.

**Review by carolinemodic — COMMENTED** [review]
Some very minor comments, seeing if I can try it out now...

File: apps/bsd/src/screens/WriteInAdjudicationScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+        }
+      } else {
+        setSelectedContestIndex((prev) => prev + 1)
+      }
+    },
+    [isLastContestSelected, makeCancelable, onAdjudicationComplete]
+  )
+
+  return (
+    <form onSubmit={goNext}>
```
Comment (benadida) [inline apps/bsd/src/screens/WriteInAdjudicationScreen.tsx:326]: oh my goodness, an actual HTML element that is not a React element? I am shook.

**Review by benadida — APPROVED** [review]
looks great.

File: apps/bsd/src/screens/WriteInAdjudicationScreen.tsx (hunk 2 above)
Comment (eventualbuddha) [reply apps/bsd/src/screens/WriteInAdjudicationScreen.tsx:326]: image

File: apps/bsd/src/screens/BallotEjectScreen.tsx [hunk 3]
```diff
@@ -100,23 +119,45 @@ const BallotEjectScreen = ({
 
   let actualElectionHash: string | undefined
 
-  for (const { interpretation } of [
-    reviewInfo.interpreted.front,
-    reviewInfo.interpreted.back,
+  for (const reviewPageInfo of [
+    {
+      side: 'front' as Side,
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/bsd/src/screens/BallotEjectScreen.tsx:124]: Yeah, this is not the best. For now I'm just going to go with this but I'll ensure we clean this stuff up after the demo.

File: apps/bsd/src/screens/WriteInAdjudicationScreen.tsx (hunk 1 above)
Comment (eventualbuddha) [reply apps/bsd/src/screens/WriteInAdjudicationScreen.tsx:188]:
```ts
  const onInput: React.FormEventHandler = useCallback(
    (event) => {
      const input = event.currentTarget as HTMLInputElement
```

**Pull request merged**